### DISCUSSION TITLE
Touchscreen rotation on wayland?

### DISCUSSION THREAD

**Discussion by realestninja** [efernau/rot8#43] [Q&A] [answered]
Hi,

if I understand correctly, then the `--touchscreen` parameter is only meant for x11, is it? 
When I use just `rot8`, it does not apply any transformation to my touchscreen matrix.

**Comment by efernau** [comment]
Hi,
rot8 work with x11 and sway.
Wayland itself is not supported, look at the open issue: https://github.com/efernau/rot8/issues/5
pull request welcome :)

**Comment by efernau** [comment]
For Sway map your input to the output device:

`$ swaymsg input <INPUTDEVICE> map_to_output <OUTPUTDEVICE>`

**Reply by realestninja** [reply, marked as answer]
Thanks, that got me into the right direction!

For anybody stumbling over this thread, about how to get it done:

1. Use `swaymsg -t get_inputs` and note the `Identifier` of your touchscreen. In my case, this is: `1386:21173:Wacom_HID_52B5_Finger`
2. Use `swaymsg -t get_outputs` and note the identifier of the screen that is used with the touchscreen. In my case: `eDP-1`
3. Use `map_to_output` but use it EXACTLY like this (inside a String):
```
swaymsg "input <INPUTDEVICE> map_to_output <OUTPUTDEVICE>"
```

In my case:
```
swaymsg "input 1386:21173:Wacom_HID_52B5_Finger map_to_output eDP-1"
```

It took me a while to figure it expects a string as a single argument!
